.github/workflows/run-tests.yml

```diff
@@ -13,14 +15,14 @@ on:
       commit:
         description: "Commit SHA"
         required: true
-      run_single_functional_test:
-        description: "Run a single functional test"
-        type: boolean
-        default: false
-      run_single_unit_test:
-        description: "Run a single unit test (INSTEAD of functional test)"
-        type: boolean
-        default: false
+      run_single_test_mode:
```

Comment:
can this be a single 'enum' like choice?
like 'test_mode' with values
'all -fund-tests'
'single-func-test'
'single-unit-test'

etc.
I guess this will greatly simplify the logic below

Reply:
I tried to make an "all-tests" option that is the default `test_mode` and is used when the action is run for a PR, but the default behavior when the action is run for a PR is to make all the `workflow_dispatch` inputs empty, so `inputs.test_mode` was blank instead of the default `all_tests` option.

There may be a way to set it though.. I agree it would make the code much less annoying